src/_pytest/mark/structures.py

```diff
@@ -360,12 +361,37 @@ def __call__(self, *args: object, **kwargs: object):
         return self.with_args(*args, **kwargs)
 
 
-def get_unpacked_marks(obj) -> List[Mark]:
-    """Obtain the unpacked marks that are stored on an object."""
+def get_marks_as_list(obj) -> List:
     mark_list = getattr(obj, "pytestmark", [])
     if not isinstance(mark_list, list):
         mark_list = [mark_list]
-    return normalize_mark_list(mark_list)
+    return mark_list
+
+
+def get_unpacked_marks(obj) -> List[Mark]:
+    """Obtain the unpacked marks that are stored on an object."""
+    return normalize_mark_list(get_marks_as_list(obj))
+
+
+def extract_mro_markers(cls):
+    if cls is None or getattr(cls, "mro_markers", []) or cls is object:
+        return
+
+    markers = {
+        str(mark): mark for mark in get_marks_as_list(cls) if mark.name != "parametrize"
```

Comment:
any mark may be used multiple times, not just parametrize,

removing parametrize completely makes things worse

never use dictionaries keyed by name to store markers

Reply:
What is the way can I filter duplicate marks?

Reply:
i believe a equality check/list membership test should work for now